bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -78,25 +79,38 @@ pub mod pallet {
 		/// To withdraw and deposit an asset.
 		type AssetTransactor: TransactAsset;
 
-		#[cfg(feature = "runtime-benchmarks")]
-		type Helper: BenchmarkHelper<Self::RuntimeOrigin>;
+		/// To charge XCM delivery fees
+		type XcmExecutor: ExecuteXcm<Self::RuntimeCall>;
 
+		/// Fee asset for the execution cost on ethereum
 		type FeeAsset: Get<Location>;
 
-		type DeliveryFee: Get<Asset>;
+		/// RemoteExecutionFee for the execution cost on bridge hub
+		type RemoteExecutionFee: Get<Asset>;
```

Comment:
Actually, I realized this hardcoding of fees on-chain is an anti-pattern. One of the major reasons for developing V2 was to move fee calculations offchain. So instead, `create_agent` and `register_token` extrinsics should take an extra fee parameter for bridgehub.

Reply:
> I'm not sure this is nessessary. I treat the RemoteExecutionFee more like a mechanism to avoid spamming.

This doesn't make sense. AH-SystemFrontend is sending an XCM to BH, and that the execution of that XCM requires fees to be paid on BH. You're even including `PaysFee(T::RemoteExecutionFee)` in the instruction.

> All the fees are actually covered by fee in ether. Including the execution cost on BH. It's the UI/relayer's responsibility to sum all the fees including and convert to ether to pay.

This does not solve the problem of how to pay for XCM execution on BH. And I don't want to hardcode that fee in the runtime.

Reply:
Make sense, I'll make change accordingly.

Reply:
Actually, I think we need to discuss this BH fee issue with Adrian next week. Maybe its too complicated for the user to figure out what fee to pass for BH. It is also problematic for governance origins, which need free execution.

In the meantime what about adding an `UnpaidExecution` in the message and modify XCM executor config in bridgehub-westend runtime to accept unpaid execution from SystemFrontendLocation.

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/63e3e6674b75c264e85ae3eab69db280ce5056e0